### ISSUE TITLE
makeTarget and source endpoint on same element 

### ISSUE THREAD

**Issue by busov** [jsplumb/jsplumb#686]
I have a div with endpoints at the bottom and I want to connect them to same div via makeTarget. Like that - I grab endpoint and drop it on the same div which has continuous anchor at the top so connection looks like a line which goes from the bottom to the top around the widget.When I try to connect them by dragging endpoint on div they connect and have same target/source element (div) and new endpoint is automatically created at the top so I can detach connection by dragging this endpoint. But when I do that programmatically - connection is rendered correctly but newly created endpoint is not attached to DOM somehow. I can see it in inspector panel when calling `jsPlumb.getAllConnection()` but it is not rendered in page.

tl;dr If I have div with id and source endpoint and also same div works as a target (created with makeTarget) can I set it up the way I could connect source endpoint from this div to same div as target?

**Comment by busov** [comment]
So my issue description is pretty general - main point us that I have widget with endpoint(s) and I need to have ability to drag them and drop connection on same widget. I am using `Continuous` anchors to add new connections dynamically. Currently after `jsPlumb.connect` I have no target endpoint (but when you drag and drop connection with mouse - endpoint will be in DOM) 

I've found this line of code https://github.com/jsplumb/jsplumb/blob/master/src/anchors.js#L223 which actually creates my problem. What is intention of it and is there a way to create endpoint when you `source === target` and anchor is `Continuous`?

@sporritt

**Comment by busov** [comment]
@sporritt can you suggest something? (main question is why we have to delete endpoint here - https://github.com/jsplumb/jsplumb/blob/master/src/anchors.js#L223 ?)

**Comment by sporritt** [comment]
i think a jsfiddle would be helpful.

**Comment by busov** [comment]
@sporritt sure, thanks for your time!

https://jsfiddle.net/boriau/1vzab7ua/ - Here you can drag endpoint from the bottom of the widget and drop it on same widget. You will see newly created endpoint and you can detach it from the top. Connection event shows endpoint in DOM and you can observe it in dev tools.

https://jsfiddle.net/boriau/Lo0gz6e7/6/ - Same use case but connected programmatically - jsPlumb connects endpoint from the bottom of widget to the top and creates connection same way as we did manually but target endpoint is not in DOM - you cant drag it and detach connection.

Endpoint is not being created because of this line in jsPlumb code - https://github.com/jsplumb/jsplumb/blob/master/src/anchors.js#L221

**Comment by busov** [comment]
^^ @sporritt code examples are added, can you take a look?

**Comment by busov** [comment]
@sporritt would be great to hear some feedback, Thanks!

**Comment by sporritt** [comment]
so the first one of these is working the way you want?

**Comment by busov** [comment]
@sporritt yes, in first example you can drag endpoint and drop it on div and everything works fine.
In second example same action was made but programmatically with jsPlumb.connect and as a result you can not see second endpoint. It is caused by this code in jsplumb - https://github.com/jsplumb/jsplumb/blob/master/src/anchors.js#L223

**Comment by sporritt** [comment]
in the first example you can also drag the endpoint again and it borks.  what would you want to happen in that case?

**Comment by busov** [comment]
@sporritt hm, in first example my use case is dragging source endpoint and dropping it on widget so then I will have this:
![image](https://user-images.githubusercontent.com/596592/34559919-7f74ee46-f14c-11e7-9792-19a8bc810c09.png)

then all I will be able to do is drag target endpoint out of widget to remove connection. source endpoint will be disabled once connection is established so there are no problems in my use-case.

**Comment by busov** [comment]
@sporritt what about second example? That is the issue :) Looks like programmatic connection results in same issue you mentioned. But it was reproduced by dragging source endpoint twice.

**Comment by busov** [comment]
just to be clear - first example is just to show that it works manually - issue is in 2nd example

**Comment by sporritt** [comment]
yup, i know the second one is the issue.  i was just asking how you felt about the fact that the first one can also fail.

**Comment by busov** [comment]
@sporritt so to sum up - do you have any suggestion or any comments regarding how to resolve second issue?

**Comment by busov** [comment]
@sporritt

**Comment by sporritt** [comment]
closing as 2.x is no longer supported. makeTarget and makeSource are not in 5.x, it uses an alternative mechanism.
